openquakeplatform/openquakeplatform/static/js/irv_viewer.js

```diff
@@ -14,1043 +14,850 @@
-            if(pdTempPILevel.hasOwnProperty(prop)) {
-                if(pdTempPILevel[prop] === foo) {
-                    return prop;
+    } else if (operator == 'Weighted multiplication') {
+        for (var v4 = 0; v4 < themeObj.length; v4++) {
+            var tempElementValue = 0;
+            var themeObjMunic = themeObj[v4].region;
+            // sum the themes
+            for (var w4 = 0; w4 < themekeys.length; w4++, ct++) {
+                var tempThemeName = themekeys[w4];
+                var themeWeightVal = themeWeightObj[tempThemeName];
+                if (tempElementValue == 0) {
+                    tempElementValue = (themeObj[v4][tempThemeName] * themeWeightVal);
+                } else {
+                    tempElementValue = tempElementValue * (themeObj[v4][tempThemeName] * themeWeightVal);
                 }
             }
+            subIndex[themeObjMunic] = tempElementValue;
         }
     }
-
-    function findSvi(pdData, svi) {
-        // Find all of the primary indicators
-        if (svi.some(function(currentValue) {
-            return (pdData.name == currentValue);
-        })) {
-            tempSviWeight = pdData.weight;
+    return subIndex;
+}
+
+function processIndicators(layerAttributes, projectDef) {
+    districts = [];
+    var allSVIThemes = [];
+    var allPrimaryIndicators = [];
+    var allRiskIndicators = [];
+
+    /////////////////////////////////////////////
+    //// Create the primary indicator objects ///
+    /////////////////////////////////////////////
+
+    var socialVulnIndex;
+    var riskIndex;
+
+    // Find all the nodes of type social vulnerability and risk index in the project def
+    for (var k in projectDef.children) {
+        if (projectDef.children[k].type == "Social Vulnerability Index") {
+            socialVulnIndex = projectDef.children[k].children;
+        } else if (projectDef.children[k].type == "Risk Index") {
+            riskIndex = projectDef.children[k].children;
         }
-        (pdData.children || []).forEach(function(currentItem) {
-            findSvi(currentItem, [svi]);
-        });
     }
 
-    function findAalWeight(pdData, aal) {
-        // Find the aal value
-        if (aal.some(function(currentValue) {
-            return (pdData.name == currentValue);
-        })) {
-            tempAalWeight = pdData.weight;
-        }
-        (pdData.children || []).forEach(function(currentItem) {
-            findAalWeight(currentItem, [aal]);
-        });
-    }
+    // process each Social Vulnerability Index nodes
+    // Get all the primary indicators
+    var ct = 0;
 
-    function utfGridClickEvent(utfGrid) {
-        utfGrid.on('click', function(e) {
-            processIndicators(e);
-            selectedGrid = e;
-        });
+    for (var i = 0; i < socialVulnIndex.length; i++) {
+        for (var e = 0 ; e < socialVulnIndex[i].children.length; e++, ct++ ) {
+            allPrimaryIndicators.push(socialVulnIndex[i].children[e].field);
```

Comment:
are primaryIndicators only including social vulnerability stuff, and not including risk stuff? If this is the case, please refactor into `allSvIndicators`